cli/waiter/subcommands/signal.py

```diff
@@ -0,0 +1,121 @@
+                for cluster, data in cluster_data.items()
+                for service in data['services']]
+    if len(services) == 0:
+        print_no_services(clusters, token)
+        return None
+    column_names = ['Service Id', 'Cluster', 'Instances', 'In-flight req.', 'Status', 'Last request', 'Current?']
+    tabular_output, sorted_services = tabulate_token_services(services, token, show_index=True, summary_table=False,
+                                                                column_names=column_names)
+    selected_service = get_user_selection(sorted_services, tabular_output)
+    selected_service_id = selected_service['service-id']
+    clusters = [clusters_by_name[selected_service['cluster']]]
+    return signal_service_id(clusters, selected_service_id)
+
+
+def map_instances_with_status(instances, status):
+    return [{'_status': status, **inst} for inst in instances]
+
+
+def get_instances_from_service_id(clusters, service_id):
+    query_result = query_service(clusters, service_id)
+    num_services = query_result['count']
+    if num_services == 0:
+        return False
+    services = map(lambda service_data: service_data['service'], query_result['clusters'].values())
+    instances = []
+    for service in services:
+        instances += map_instances_with_status(service['instances']['active-instances'], 'active')
+    return instances
+
+def signal(clusters, args, _, __):
+    guard_no_cluster(clusters)
+
+    token_or_service_id_or_instance_id = args.pop('token-or-service-id-or-instance-id')
+
+    signal_destination = args.pop('signal_destination')
+
+    instance_id = None
+    if signal_destination == Destination.TOKEN:
+        selected_instance = signal_token(clusters, token_or_service_id_or_instance_id)
+        if selected_instance != None:
+            instance_id = selected_instance.get('id', None)
+        else:
+            return 0
+    elif signal_destination == Destination.SERVICE_ID:
+        selected_instance = signal_service_id(clusters, token_or_service_id_or_instance_id)
+        if selected_instance != None:
+            instance_id = selected_instance.get('id', None)
+        else:
+            return 0
+    elif signal_destination == Destination.INSTANCE_ID:
+        instance_id = token_or_service_id_or_instance_id
+    
+    signal_type = args.pop('signal_type')
+    timeout_secs = args['timeout']
+    success = False
+
+    if signal_type == Signal.SIGKILL.value:
+        success = process_signal_request(clusters, signal_type, instance_id, timeout_secs)
+    elif signal_type == Signal.SIGTERM.value:
+        success = process_signal_request(clusters, signal_type, instance_id, timeout_secs)
+
+    return 0 if success else 1
+
+
+def register(add_parser):
+    """Adds this sub-command's parser and returns the action function"""
+    parser = add_parser('signal', help='sends signal to instance')
+    parser.add_argument('signal_type', help='type of signal to send to instance')
```

Comment:
Is this required? If not, we should have a default value.

We should also consider using named constants, e.g. sigkill, sigterm or aliases like hard-delete, soft-delete (and later eject, expire).